packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/spec-patchers.tsx

```diff
@@ -322,11 +321,13 @@ export function patchQuickSearchService(framework: FrameworkProvider) {
               framework.get(DocsQuickSearchSession),
               framework.get(CreationQuickSearchSession),
               (query: string) => {
-                if (
-                  (query.startsWith('http://') ||
-                    query.startsWith('https://')) &&
-                  resolveLinkToDoc(query) === null
-                ) {
+                const isLink =
+                  query.startsWith('http') &&
+                  (query.startsWith('://', 4) || query.startsWith('s://', 4));
```

Comment:
is this any better than startsWith('http://')?

Reply:
No need to compare again from the starting position

Reply:
I think this is over optimization. You try to optimize one function call into two.

Reply:
Ok, back